model/riscv_insts_zce_ext.sail

```diff
@@ -0,0 +1,1021 @@
+  <-> "c.zext.b" ^ spc() ^ creg_name(rsd)
+
+/* ****************************************************************** */
+union clause ast = C_SEXT_B : (cregidx)
+
+//$[wavedrom FUNCT3 _ SRCDST FUNCT2 C.SEXT.B C1]
+mapping clause encdec_compressed = C_SEXT_B(rsd)
+  <-> 0b100 @ 0b111 @ rsd : cregidx @ 0b11 @ 0b001 @ 0b01
+
+function clause execute (C_SEXT_B(rsd)) = {
+  let rsdc = creg2reg_idx(rsd);
+  
+  X(rsdc) = EXTS(X(rsdc)[7..0]);
+  RETIRE_SUCCESS
+}
+
+mapping clause assembly = C_SEXT_B(rsd)
+  <-> "c.sext.b" ^ spc() ^ creg_name(rsd)
+/* ****************************************************************** */
+union clause ast = C_ZEXT_H : (cregidx)
+
+//$[wavedrom FUNCT3 _ SRCDST FUNCT2 C.ZEXT.H C1]
+mapping clause encdec_compressed = C_ZEXT_H(rsd)
+  <-> 0b100 @ 0b111 @ rsd : cregidx @ 0b11 @ 0b010 @ 0b01
+
+function clause execute (C_ZEXT_H(rsd)) = {
+  let rsdc = creg2reg_idx(rsd);
+  
+  X(rsdc) = EXTZ(X(rsdc)[15..0]);
+  RETIRE_SUCCESS
+}
+
+mapping clause assembly = C_ZEXT_H(rsd)
+  <-> "c.zext.h" ^ spc() ^ creg_name(rsd)
+
+/* ****************************************************************** */
+union clause ast = C_SEXT_H : (cregidx)
+
+//$[wavedrom FUNCT3 _ SRCDST FUNCT2 C.SEXT.H C1]
+mapping clause encdec_compressed = C_SEXT_H(rsd)
+  <-> 0b100 @ 0b111 @ rsd : cregidx @ 0b11 @ 0b011 @ 0b01
+
+function clause execute (C_SEXT_H(rsd)) = {
+  let rsdc = creg2reg_idx(rsd);
+  
+  X(rsdc) = EXTS(X(rsdc)[15..0]);
+  RETIRE_SUCCESS
+}
+
+mapping clause assembly = C_SEXT_H(rsd)
+  <-> "c.sext.h" ^ spc() ^ creg_name(rsd)
+/* ****************************************************************** */
+union clause ast = C_ZEXT_W : (cregidx)
+
+//$[wavedrom FUNCT3 _ SRCDST FUNCT2 C.ZEXT.W C1]
+mapping clause encdec_compressed = C_ZEXT_W(rsd)
+  <-> 0b100 @ 0b111 @ rsd : cregidx @ 0b11 @ 0b100 @ 0b01
+
+function clause execute (C_ZEXT_W(rsd)) = {
+  let rsdc = creg2reg_idx(rsd);
+  
+  X(rsdc) = EXTZ(X(rsdc)[31..0]);
+  RETIRE_SUCCESS
+}
+
+mapping clause assembly = C_ZEXT_W(rsd)
+  <-> "c.zext.w" ^ spc() ^ creg_name(rsd)
+
+/* ****************************************************************** */
+
+/* ------------------------------------------------------------------ */
+/* Sail Model for the c.mul encoding group                            */
+/* ------------------------------------------------------------------ */
+
+/* the assembly abstract syntax tree (AST) clause for the CMUL instructions */
+union clause ast = C_MUL : (cregidx,cregidx)
+
+/* the encode/decode mapping between AST elements and 16-bit half words */
+
+//$[wavedrom FUNCT3 _ SRCDST FUNCT2 C.MUL C1]
+mapping clause encdec_compressed = C_MUL(rsd, rs2)
+  <-> 0b100 @ 0b111 @ rsd : cregidx @ 0b00 @ rs2 : cregidx @ 0b01
+
+function clause execute (C_MUL(rsd,rs2)) = {
+  let rd = creg2reg_idx(rsd);
+  let rs = creg2reg_idx(rs2);
+  
+  let result_wide = to_bits(2 * sizeof(xlen), signed(X(rd)) * signed(X(rs)));
+  X(rd) = result_wide[(sizeof(xlen) - 1) .. 0];
+  RETIRE_SUCCESS
+}
```

Comment:
Why don't you map this to the existing implementation of mul in a similar way to add c.add and other existing existing compressed instructions ? https://github.com/riscv/sail-riscv/blob/72b2516d10d472ac77482fd959a9401ce3487f60/model/riscv_insts_cext.sail#L619